client/ayon_core/hosts/maya/plugins/publish/collect_look.py

```diff
@@ -439,14 +372,87 @@ def collect(self, instance):
 
         self.log.debug("Collected look for %s" % instance)
 
+    def collect_file_nodes(self, look_sets):
+        """Get the entire node chain of the look sets and return file nodes
+
+        Arguments:
+            look_sets (List[str]): List of sets and shading engines relevant
+                to the look.
+
+        Returns:
+            List[str]: List of file node names.
+
+        """
+
+        shader_attrs = [
+            "surfaceShader",
+            "volumeShader",
+            "displacementShader",
+            "aiSurfaceShader",
+            "aiVolumeShader",
+            "rman__surface",
+            "rman__displacement"
+        ]
+
+        # Get all material attrs for all look sets to retrieve their inputs
+        existing_attrs = []
+        for look_set in look_sets:
+            for attr in shader_attrs:
+                if cmds.attributeQuery(attr, node=look_set, exists=True):
+                    existing_attrs.append("{}.{}".format(look_set, attr))
+
+        materials = cmds.listConnections(existing_attrs,
+                                         source=True,
+                                         destination=False) or []
+
+        self.log.debug("Found materials:\n{}".format(materials))
+
+        # Get the entire node chain of the look sets
+        # history = cmds.listHistory(look_sets, allConnections=True)
+        # if materials list is empty, listHistory() will crash with
+        # RuntimeError
+        history = set()
+        if materials:
+            history = set(cmds.listHistory(materials, allConnections=True))
```

Comment:
history is already the empty set, guess we can just add the element into the set instead.
```suggestion
            history.add(cmds.listHistory(materials, allConnections=True))
```

Reply:
And the update is somewhat faster:
```python
>>> print(timeit.timeit(stmt='a = set([i for i in range(1,100)])', globals=globals()))
4.729680200049188
>>> a =set()
>>> print(timeit.timeit(stmt='a.update([i for i in range(1,100)])', globals=globals()))
3.7812449999619275
```